libs/types/src/tokens.rs

```diff
@@ -187,8 +187,8 @@ where
 )]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct TrancheCurrency {
-	pub(crate) pool_id: PoolId,
-	pub(crate) tranche_id: TrancheId,
+	pub pool_id: PoolId,
```

Comment:
any objections to open these fields visibility? by having them `pub(crate)` I would have to import he `TrancheCurrencyT` trait to call `generate` to build an instance of this type which is quite an overkill imo. let me know if I am not overseeing any requirement 👍